packages/app/components/checkout/checkout-form.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from "react";
+import { useState, useMemo } from "react";
+
+import {
+  useStripe,
+  useElements,
+  Elements,
+  PaymentElement,
+  LinkAuthenticationElement,
+} from "@stripe/react-stripe-js";
+
+import { Button } from "@showtime-xyz/universal.button";
+import { Checkbox } from "@showtime-xyz/universal.checkbox";
+import { useIsDarkMode } from "@showtime-xyz/universal.hooks";
+import { Text } from "@showtime-xyz/universal.text";
+import { View } from "@showtime-xyz/universal.view";
+
+import { Logger } from "app/lib/logger";
+
+import { stripePromise } from "./stripe";
+
+export function CheckoutForm({ clientSecret }: { clientSecret: string }) {
+  const isDark = useIsDarkMode();
+  const stripeOptions = useMemo(
+    () =>
+      ({
+        clientSecret,
+        appearance: {
+          theme: isDark ? "night" : "stripe",
+        },
+      } as const),
+    [clientSecret, isDark]
+  );
+
+  return stripeOptions?.clientSecret ? (
+    <Elements stripe={stripePromise} options={stripeOptions}>
+      <CheckoutFormStripe />
+    </Elements>
+  ) : null;
+}
+
+const CheckoutFormStripe = () => {
+  const stripe = useStripe();
+  const elements = useElements();
+  const [setAsDefaultPaymentMethod, setSetAsDefaultPaymentMethod] =
+    useState(true);
+
+  const [email, setEmail] = useState("");
+  const [message, setMessage] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleSubmit = async (e: any) => {
+    e.preventDefault();
+
+    if (!stripe || !elements) {
+      return;
+    }
+
+    setIsLoading(true);
+
+    const { error } = await stripe.confirmPayment({
+      elements,
+      confirmParams: {
+        return_url:
+          window.location.origin +
+          "/checkout-return?setAsDefaultPaymentMethod=" +
+          setAsDefaultPaymentMethod,
```

Comment:
I know we're not using this form in our native apps, but checkout-form.tsx is not suffixed with `.web.tsx`. In theory, this would be loaded on native. `window.location` is not available. We should either make that an web only component or add some gating with Platform.OS

Reply:
yeah, I considered it. This component is not imported (in root-navigator) so won't be included in bundle. But yeah better to add the suffix. Will do!

Reply:
@hirbod adding some conditionals for now. When we add checkout on native, it'll be a complete rewrite, we'll not use stripe. We can spend more time later cleaning some stuff. Ignoring it for now.

Reply:
Ok, makes sense.